### PR TITLE
Add tests around `persist`

### PR DESCRIPTION
**Pull request by keithluchtel**
Adds tests to validate reading data on initialize and writing data on actions. Adds additional test to validate delay configuration.

### CODE REVIEW

File: test-utils/storage.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  setItem: (key: string, data: string) => {
+    store[key] = data;
+    return Promise.resolve();
+  },
+  getItem: (key: string) => {
+    const value = store[key];
+    return Promise.resolve(value);
+  },
+  clearStore: () => {
+    for (const key in store) delete store[key];
```
Comment (carlos-kelly) [inline test-utils/storage.ts:14]: This looks good and works for tests, the only thing you could perhaps leverage is a `Map` which is typical for a quick in-memory store.

File: test-utils/waitFor.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export function waitFor(conditionFunction: () => boolean) {
+  const poll = (resolve: (value: unknown) => void) => {
+    if (conditionFunction()) resolve(null);
+    else setTimeout(() => poll(resolve), 100);
+  };
```
Comment (carlos-kelly) [inline test-utils/waitFor.ts:5]: At some point after x iterations should we reject? Perhaps a `failAfter` param?

**Review by carlos-kelly — APPROVED** [review]
Great work and use of vitest. No blocking suggestions to merge.

File: test-utils/storage.ts (hunk 1 above)
Comment (keithluchtel) [reply test-utils/storage.ts]: done

File: test-utils/waitFor.ts (hunk 2 above)
Comment (keithluchtel) [reply test-utils/waitFor.ts:13]: done

**Pull request merged**